objectherkenning_openbare_ruimte/convert_dataset_pipeline/source/equirectangular_to_cubemap_converter.py

```diff
@@ -0,0 +1,752 @@
+
+    def visualize_annotations_on_equirectangular_image(self, img_path: str) -> None:
+        """
+        Visualize YOLO annotations on the original equirectangular image.
+
+        Parameters
+        ----------
+        img_path : str
+            The name of the image file to visualize annotations for.
+
+        Returns
+        -------
+        None
+        """
+
+        image_file_path = os.path.join(self.input_path, img_path)
+        annotation_file_path = os.path.join(
+            self.input_path, img_path.split(".")[0] + ".txt"
+        )
+
+        image = cv2.imread(image_file_path)
+        if image is None:
+            logging.error(f"Image file {image_file_path} not found.")
+            return
+
+        EquirectangularToCubemapConverter._draw_annotations_from_file(
+            annotation_file_path, image, draw_thickness=2
+        )
+
+        annotated_folder = os.path.join(self.input_path, "input_annotated")
+        if not os.path.exists(annotated_folder):
+            os.makedirs(annotated_folder)
+
+        visualized_img_path = os.path.join(
+            annotated_folder, img_path.split(".")[0] + "_annotated.png"
+        )
+        cv2.imwrite(visualized_img_path, image)
+
+        logging.info(f"Annotated equirectangular image saved to {visualized_img_path}")
+
+    def convert_image(self, img_path: str) -> None:
+        """
+        Convert an equirectangular image to a set of six cubic faces using py360convert.e2c.
+
+        Parameters
+        ----------
+        img_path : str
+            Path to the input image file.
+
+        Returns
+        -------
+        None
+
+        """
+
+        img = os.path.join(self.input_path, img_path)
+        img = cv2.imread(img)
+
+        try:
+            if img is None:
+                raise FileNotFoundError(f"File {img} is empty or corrupted")
+        except FileNotFoundError as e:
+            logging.error(str(e))
+            return
+
+        try:
+            front, right, back, left, top, bottom = p3c.e2c(
+                img, face_w=self.face_width, mode="bilinear", cube_format="list"
+            )
+
+            base_name = os.path.basename(img_path)
+            folder = os.path.splitext(base_name)[0]
+            directory = os.path.join(self.output_path, folder)
+            if not os.path.exists(directory):
+                os.makedirs(directory)
+
+            cv2.imwrite(f"{directory}/front.png", front)
+            cv2.imwrite(f"{directory}/right.png", right)
+            cv2.imwrite(f"{directory}/back.png", back)
+            cv2.imwrite(f"{directory}/left.png", left)
+            cv2.imwrite(f"{directory}/top.png", top)
+            cv2.imwrite(f"{directory}/bottom.png", bottom)
+
+            logging.info(f"Processed image: {img_path}.")
+        except Exception as e:
+            logging.error(f"Error processing image {img_path}: {e}")
+
+    def process_annotations(self, img_path: str) -> None:
```

Comment:
Add some inline comments in this function to indicate what each part is doing (for-loops, if-statements).